spark/src/main/scala/geotrellis/spark/io/s3/RasterRDDReader.scala

```diff
@@ -0,0 +1,151 @@
+package geotrellis.spark.io.s3
+
+import geotrellis.spark._
+import geotrellis.raster.Tile
+import geotrellis.spark.io._
+import geotrellis.spark.io.index.KeyIndex
+import geotrellis.spark.utils.KryoSerializer
+import org.apache.spark.SparkContext
+import com.amazonaws.auth.AWSCredentialsProvider
+import com.amazonaws.services.s3.model.{ListObjectsRequest, ObjectListing}
+import com.typesafe.scalalogging.slf4j._
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+abstract class RasterRDDReader[K: ClassTag] extends LazyLogging {
+  import RasterRDDReader._
+
+  def setFilters(filterSet: FilterSet[K], kb: KeyBounds[K], ki: KeyIndex[K]): Seq[(Long, Long)]
+
+  def read(
+    s3client: () => S3Client,
+    layerMetaData: S3LayerMetaData,
+    keyBounds: KeyBounds[K],
+    keyIndex: KeyIndex[K])
+  (layerId: LayerId, filterSet: FilterSet[K])
+  (implicit sc: SparkContext): RasterRDD[K] = {
+    val bucket = layerMetaData.bucket
```

Comment:
Got it. That works. I guess we should make them all the same? @moradology is that something you could do as part of the refactor PR?